### DISCUSSION TITLE
Uninstalling HAA from Sonoff Mini R2 to install  original Sonoff/Tasmota/ESPHome etc.

### DISCUSSION THREAD

**Discussion by Hl2run** [RavenSystem/esp-homekit-devices#2066] [Q&A] [answered]
Hello,

I installed HAA via OTA on my Sonoff Mini R2. Currently running 11.8.

Now I would like to install a different firmware instead of HAA (love your work, but I need to put this into Home Assistant) and I cannot figure out how to flash something else over-the-air on this device.

Can anyone suggest a way?

Thank you.

**Comment by hkuehl** [comment, marked as answer]
Short answer is: you can’t. There are other threads that already complained about this. Only wired reset works.
